pkg/R/DataFrame.R

```diff
@@ -52,21 +53,108 @@ dataFrame <- function(sdf) {
 #' printSchema(df)
 #'}
 
-setGeneric("printSchema", function(df) { standardGeneric("printSchema") })
+setGeneric("printSchema", function(x) { standardGeneric("printSchema") })
 
 setMethod("printSchema",
-          signature(df = "DataFrame"),
-          function(df) {
-            sdf <- df@sdf
-            schemaString <- callJMethod(sdf, "printSchema")
+          signature(x = "DataFrame"),
+          function(x) {
+            schemaString <- callJMethod(x@sdf, "printSchema")
             cat(schemaString)
           })
 
+#' Get schema object
+#' 
+#' Returns the schema of this DataFrame as a list of named lists. Each named
```

Comment:
@davies Just committed wrapper functions for `structType` and `structField`. @shivaram your suggestion to look at `jobj.R` was a big help. I ended up using the same approach to basically replicate both types as well as their methods on Spark side, and they both have `print` methods in R as well. 

I think this approach is very extensible as well, and should come in handy as we need to include more of the Spark DataTypes.